src/excited_workflow/carbon_tracker.py

```diff
@@ -0,0 +1,266 @@
+"""Train carbon tracker datasets."""
+
+import datetime
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import onnxmltools
+import pandas as pd
+import pycaret.regression
+import xarray as xr
+from dask.distributed import Client
+from skl2onnx.common.data_types import DoubleTensorType
+
+import excited_workflow
+from excited_workflow.source_datasets import datasets
+
+
+def merge_datasets(desired_data: list[str], target: Path) -> xr.Dataset:
+    """Merge datasets onto one common grid.
+
+    Args:
+        desired_data: list of desired datasets
+        freq: time frequency that is desired
+        target: target grid dataset
+
+    Returns:
+        Dataset of input variables.
+    """
+    ds_cb = xr.open_dataset(target)
+    ds_cb = excited_workflow.utils.convert_timestamps(ds_cb)
+    ds_input = xr.merge(
+        [
+            datasets[name].load(freq="monthly", target_grid=ds_cb)
+            for name in desired_data
+        ]
+    )
+
+    return ds_input
+
+
+def mask_region(regions: Path, target: Path, ds_input: xr.Dataset) -> xr.Dataset:
+    """Limit data to a region and time slice.
+
+    Args:
+        regions: path to regions file.
+        target: path to target dataset.
+        ds_input: input dataset.
+
+    Returns:
+        Dataset of masked to North America.
+    """
+    ds_regions = xr.open_dataset(regions)
+    ds_cb = xr.open_dataset(target)
+    ds_cb = excited_workflow.utils.convert_timestamps(ds_cb)
+    ds_merged = xr.merge(
+        [
+            ds_cb[["bio_flux_opt"]],
+            ds_regions["transcom_regions"],
+            ds_input,
+        ]
+    )
+    time_region_na = {
+        "time": slice("2000-01", "2019-12"),
+        "latitude": slice(15, 60),
+        "longitude": slice(-140, -55),
+    }
+    ds_na = ds_merged.sel(time_region_na)
+    ds_na = ds_na.compute()
+    ds_na = ds_na.where(ds_merged["transcom_regions"] == 2)
+
+    return ds_na
+
+
+def create_groups(ds: xr.Dataset, number: int) -> pd.DataFrame:
+    """Create dataframe with different groups.
+
+    Args:
+        ds: Dataset to split into groups.
+        number: number of groups.
+
+    Returns:
+        Dataframe with groups column.
+    """
+    df_train = ds.to_dataframe().dropna()
+    groups = number
+    splits = np.array_split(df_train, groups)
+    for i in range(len(splits)):
+        splits[i]["group"] = i
+
+    df_train = pd.concat(splits)
+    return df_train
+
+
+def create_df(df: pd.DataFrame, x_keys: list[str], y_key: str) -> pd.DataFrame:
+    """Create a dataframe for training.
+
+    Args:
+        df: dataframe which needs to be converted for training.
+        x_keys: list of input variables.
+        y_key: target variable name.
+
+    Returns:
+        Dataframe for training.
+    """
+    df_pycaret = df[x_keys + [y_key]]
+    df_reduced = df_pycaret[::10]
+
+    return df_reduced
+
+
+def train_model(
```

Comment:
I think it's better to name this function differently, something like `groupwise_cross_validation`. Because the model training here is just to perform CV.